### PR TITLE
Update DG Storage component and add Edit implementation 

### LINKED ISSUE #43: Update Storage component in DG
- Update the prose explaining the component
- Update the UML class diagram

### LINKED ISSUE #44: Add edit command implementation to DG
- Describe the implementation

- Create sequence diagram for the execution flow

- List out steps and execution behavior for the edit command

### PR DESCRIPTION
**Pull request by Silvernitro**
- Update the `Storage` component class diagram
- Add implementation details for the edit command
- Add sequence diagram for the execution of edit commands

Note: To view the new UML diagrams, you can view the png files in the PR instead of reading the puml file. Click the button near the top right of the png file that says "rich diff"

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -133,6 +133,52 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+* Module Code
+* Modular Credits
+* Tags
+
+As with all operations in GradPad, the `EditCommand` class handles the execution of edit operations.
+The `EditCommandParser` class helps to parse a user's input before creating the correct edit command.
+
+GradPad uses the `EditCommandDescriptor` class to facilitate edit operations. An `EditCommandDescriptor` is
+a temporary bridge that holds the newly-edited fields of a module. You can set the `ModuleCode`,
+`ModularCredit`, and `Tags` of an `EditCommandDescriptor`. It is also fine to omit any of them, which is important
```
Comment (yan-soon) [inline docs/DeveloperGuide.md]: Did we refactor the EditCommandDescriptor class to EditModuleDescriptor class?

File: docs/diagrams/EditSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,90 @@
+@startuml
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":GradPadParser" as GradPadParser LOGIC_COLOR
+participant ":EditCommandParser" as EditCommandParser LOGIC_COLOR
+participant ":ParserUtil" as ParserUtil LOGIC_COLOR
+participant ":EditCommandDescriptor" as EditCommandDescriptor LOGIC_COLOR
```
Comment (yan-soon) [inline docs/diagrams/EditSequenceDiagram.puml]: Same comment as above regarding the EditCommandDescriptor.

**Review by yan-soon — COMMENTED** [review]
Steps are clear and the Sequence Diagram for the Storage component and Edit feature looks great!

**Comment by Silvernitro** [comment]
@yan-soon typos have been fixed; thanks so much for catching them!

**Comment by yan-soon** [comment]
LGTM!

**Review by yan-soon — APPROVED** [review]

**Pull request merged**